ArtOfIllusion/src/artofillusion/material/LinearMaterialMapping.java

```diff
@@ -45,7 +45,8 @@ public String getName()
     return "Linear";
   }
 
-  public static boolean legalMapping(Object3D obj, Material mat)
+  @Override
+  public boolean legalMapping(Object3D obj, Material mat)
```

Comment:
In this case, I think that there may be, mostly because this _is_ intended as a public API, a designed extension point. It was always intended that plugins would add other mapping types, and I have at least one in the back of my mind that would do so.

I'd expect an intentional API to try to be as natural as possible. (What some call the 'principle of least surprise')

Think of it from the viewpoint of someone developing a plugin or script. The natural thing to do, if you don't realize that `legalMapping()` is overridden by every subclass, (or don't realize all of the implications) is to call it in `instance.legalMapping()` style. We could argue about whether Java should allow that to happen, but it _does_ allow it. The developer would probably get away with it for a while, especially with a Material mapping. As soon as someone adds a new type of mapping, however, things will break in unexpected, probably hard to de-bug ways.

I'm not saying that we need to change this tomorrow, but we should consider it while we still have a plugin ecosystem that will mostly not even notice.

What I would like to do is start a list of breaking API changes/improvements that would be nice to have. We can add to them as we continue reviewing the code and cleaning up lint points, etc. Once we have a full list, we could make all of the changes in one version increment, so that we have a well-defined API version change.